src/SearchIndex/index.ts

```diff
@@ -266,3 +267,23 @@ const composeUpserts = (
     };
   });
 };
+
+// similar to atlas admin getFacetKeys but this one is not
+// specific to taxonomy type
+export const getFacetKeysFromResponseFacets = (facets: FacetOption[]): string[] => {
```

Comment:
can we move this fn to Query/utils and also add a unit test for this

Reply:
or not! see comment above

Reply:
yes, for comment above, we don't have the taxonomy at hand at query time and acquiring it would either create a large `async`/`await` chain or require passing it through many functions, but I think storing it locally or storing the result of `getFacetKeys` would be useful. I think this is where some stuff about organization I mentioned in Engineering Sync was coming up, but I'll take another crack at this and write everything down.